esp_custom_part_ota/include/esp_custom_part_ota.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdbool.h>
+#include <esp_err.h>
+#include <esp_idf_version.h>
+#include "esp_partition.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void *esp_custom_part_ota_handle_t;
+
+typedef struct {
+    const esp_partition_t *update_partition;
+    const esp_partition_t *backup_partition;
+} esp_custom_part_ota_cfg_t;
+
+
+/**
+ * @brief   Commence an OTA update writing to the specified partition.
+ *
+ * @param config    Configuration data for the OTA process
+ *
+ * @return
+ *    - NULL    On failure
+ *    - esp_custom_part_ota_handle_t handle
+*/
+esp_custom_part_ota_handle_t esp_custom_part_ota_begin(esp_custom_part_ota_cfg_t config);
+
+
+/**
+ * @brief   Write OTA update data to partition
+ *
+ * This function can be called multiple times as
+ * data is received during the OTA operation. Data is written
+ * sequentially to the partition.
+ *
+ * @param ctx   esp_custom_part_ota_handle_t
+ * @param data  Data buffer to write
+ * @param size  Size of the data buffer in bytes
+ *
+ * @return
+ *      - ESP_OK
+ *      - ESP_ERR_INVALID_ARG
+ *      - lower level esp_partition error codes.
+*/
+esp_err_t esp_custom_part_ota_write(esp_custom_part_ota_handle_t handle, const void *data, size_t size);
+
+
+/**
+ * @brief   Finish OTA update
+ *
+ * @param ctx   esp_custom_part_ota_handle_t
+ *
+ * @return
+ *      - ESP_OK
+ *      - ESP_ERR_INVALID_ARG
+*/
+esp_err_t esp_custom_part_ota_end(esp_custom_part_ota_handle_t handle);
+
+
+/**
+ * @brief   Abort the OTA update process and free the handle
+ *
```

Comment:
We can mention here that After aborting the OTA, you need to call `... restore_ota ...` to restore the backup if available